datatypes-jws/src/commonMain/kotlin/at/asitplus/crypto/datatypes/jws/JwsHeader.kt

```diff
@@ -45,20 +45,50 @@ data class JwsHeader(
     @Serializable(with = InstantLongSerializer::class)
     val expiration: Instant? = null,
     @SerialName("jwk")
-    val jsonWebKey: JsonWebKey? = null
+    val jsonWebKey: JsonWebKey? = null,
+    @SerialName("jku")
+    val jsonWebKeySetUrl: String? = null,
+    /**
+     * OID4VP: Verifier Attestation JWT, used to authenticate a Verifier, by providing a JWT signed by a trusted
+     * third party. May be parsed as a [JwsSigned], with [JsonWebToken] as the payload.
+     */
+    @SerialName("jwt")
+    val attestationJwt: String? = null,
 ) {
 
     fun serialize() = jsonSerializer.encodeToString(this)
 
+    /**
+     * Tries to compute a public key in descending order from [jsonWebKey], [keyId],
+     * or [certificateChain], and takes the first success or null.
+     */
+    val publicKey: CryptoPublicKey? by lazy {
+        jsonWebKey?.toCryptoPublicKey()?.getOrNull()
+            ?: keyId?.let { runCatching { CryptoPublicKey.fromDid(it) } }?.getOrNull()
+            ?: certificateChain
+                ?.firstNotNullOfOrNull {
+                    runCatching {
+                        X509Certificate.decodeFromTlv(Asn1Element.parse(it) as Asn1Sequence).publicKey
+                    }.getOrNull() ?: runCatching {
+                        X509Certificate.decodeFromTlv(Asn1Element.parse(it.decodeX5c()) as Asn1Sequence).publicKey
+                    }.getOrNull()
+                }
+    }
+
+    private fun ByteArray.decodeX5c() = decodeToString()
```

Comment:
A function like that should not be just a private utility in KMP-Crypto.
It would make more sense to add it to the certificate class, for example, because it provides added value in general.